components/authz-service/server/v2/project_update_manager.go

```diff
@@ -285,45 +284,20 @@ func (w *workflowInstance) PercentageComplete() float64 {
 		return 1.0
 	}
 
-	percentComplete := 0.0
-	domainServicesUpdateInstance, err := w.GetUpdateDomainServicesInstance()
-	if err == cereal.ErrWorkflowInstanceNotFound {
-		return percentComplete
-	}
-
-	domainServices := domainServicesUpdateInstance.ListSubWorkflows()
-	if len(domainServices) == 0 {
+	domainEstimates := w.collectAllDomainServiceEstimates()
+	if len(domainEstimates) == 0 {
 		return 1.0
 	}
 
-	longestEstimatedTimeComplete := time.Time{}
-	for _, d := range domainServicesUpdateInstance.ListSubWorkflows() {
-		subWorkflow, err := domainServicesUpdateInstance.GetSubWorkflow(d)
-		if err != nil {
-			logrus.WithError(err).Errorf("failed to get subworkflow for %q", d)
-			continue
-		}
-		payload := project_update_tags.DomainProjectUpdateWorkflowPayload{}
-		if subWorkflow.IsRunning() {
-			if err := subWorkflow.GetPayload(&payload); err != nil {
-				logrus.WithError(err).Errorf("failed to get payload for %q", d)
-				continue
-			}
-
-			if !payload.MergedJobStatus.Completed {
-				estimatedTime := time.Unix(payload.MergedJobStatus.EstimatedEndTimeInSec, 0)
-				if estimatedTime.After(longestEstimatedTimeComplete) {
-					percentComplete = float64(payload.MergedJobStatus.PercentageComplete)
```

Comment:
Here is where the bug was. We should have been setting `longestEstimatedTimeComplete = estimatedTime`.